circular Queue 리팩토링 했습니다!

myangw opened the pull request:
- Create queue structure.
- Add circular queue functions.

myangw committed:
Add circular queue

- Create queue structure.
- Add circular queue functions.

kwangilcho commented on stack and queue/circular_queue.c:
```diff
@@ -0,0 +1,172 @@
+}
+
+int IsEmpty(struct Queue *queue) {
+    if (queue->size == 0) {
```
```c
return queue->size == 0;
```
비교 연산 한 번으로.

kwangilcho commented on stack and queue/circular_queue.c:
```diff
@@ -0,0 +1,172 @@
+}
+
+int IsFull(struct Queue *queue) {
+    if (queue->size == queue->capacity) {
```
```c
return queue->size == queue->capacity;
```

kwangilcho commented on stack and queue/circular_queue.c:
```diff
@@ -0,0 +1,172 @@
+}
+
+int Succ(int index, struct Queue *queue) {
+    if (++index == queue->capacity)
```
```c
return ++index == queue->capacity ? 0 : index;
```
삼항 연산 한 번으로

kwangilcho commented on stack and queue/circular_queue.c:
```diff
@@ -0,0 +1,172 @@
+    return index;
+}
+
+void PrintAll(struct Queue *queue) {
```
큐의 size를 알고 있으므로 size를 활용하면 훨씬 간결하게 구현할 수 있지 않을까요?

kwangilcho commented on stack and queue/circular_queue.c:
```diff
@@ -0,0 +1,172 @@
+    printf("f: %d, r: %d\n", front, rear);
+}
+
+struct Queue *Dequeue(struct Queue *queue) {
```
return type을 void로 해도 괜찮지 않을까요?

kwangilcho commented on stack and queue/circular_queue.c:
```diff
@@ -0,0 +1,172 @@
+    return queue;
+}
+
+struct Queue *Enqueue(struct Queue *queue, int new) {
```
return type을 void로 해도 괜찮지 않을까요?

kwangilcho reviewed (commented):
잘 구현하셨네요! 💯
비교 연산을 조금 더 간결하게 만들어보고,
자료구조가 가지고 있는 메타데이터를 활용해보면 좋겠어요.

myangw committed:
Update circular Queue

- IsEmpty, IsFull, Succ 함수 비교연산 간결하게 변경
- PrintAll 함수 queue->size 이용하여 불필요한 분기문 제거
